fdbserver/include/fdbserver/LogSystem.h

```diff
@@ -842,6 +845,15 @@ struct LogPushData : NonCopyable {
 	Tag chooseRouterTag() {
 		return savedRandomRouterTag.present() ? savedRandomRouterTag.get() : logSystem->getRandomRouterTag();
 	}
+
+	void getPushLocations(std::vector<Tag> const& tags, std::vector<int>& locations, bool allLocations = false) {
```

Comment:
This is not using `tags`, which can be a much smaller set than the set `storedLocations`. So this doesn't seem right.

The idea of `logSystem->getPushLocations()` is to find the tlogs that should receive the tags. We don't want to send the message to a tlog that doesn't have to receive it.

Reply:
It is using prev_tags , which is a member of the class. prev_tags was used in the original call, so I think the code is still correct. However, the parameter should be used (or dropped). I'll update this.

Reply:
As I thought more about it, this is called by `writeTypedMessage()`, which is called almost on every mutation, e.g., in `writeMutation()`. So I think we need to optimize this to avoid sending every mutation to all tlogs.